src/libs/auth.js

```diff
@@ -37,6 +37,10 @@ export const hasBillingScope = () => {
   return getAuthInstance().currentUser.get().hasGrantedScopes('https://www.googleapis.com/auth/cloud-billing')
 }
 
+const registeredAndAcceptedTos = (oldState, state) => {
+  return (oldState.registrationStatus !== 'registered' && state.registrationStatus === 'registered' && state.acceptedTos) ||
+    (state.registrationStatus === 'registered' && state.acceptedTos && !oldState.acceptedTos)
+}
```

Comment:
I don't actually know if _all_ the changes below were necessary, but I am assuming that if the user hasn't accepted the ToS they shouldn't be able to do much. I can try to narrow down which ones actually need this check tomorrow.

Reply:
I will continue to look at this in the follow-up PR.

Reply:
Another note for context: we were hitting a problem where the 'getFenceStatus' call was happening when the user became registered, before ToS was accepted. Since ToS is now required, this resulted in a 401, whereas it used to result in a 403. The 401 caused the user to be logged out due to [this code](https://github.com/DataBiosphere/terra-ui/blob/dev/src/libs/error.js#L7), failing the test since we did not get to the expected Workspaces list.

Then once I fixed the 'getFenceStatus' issue, the test appeared to run correctly, but still failed due to the NpsSurvey unprotected AJAX call.

Reply:
Ah, this context was helpful; thank you.